ipcl/mod_exp.cpp

```diff
@@ -64,6 +64,127 @@ bool isHybridOptimal() {
 }
 
 #ifdef IPCL_USE_QAT
+#include <omp.h>
+static std::vector<BigNumber> heQatBnModExp_MT(
+    const std::vector<BigNumber>& base, const std::vector<BigNumber>& exponent,
+    const std::vector<BigNumber>& modulus, unsigned int batch_size) {
+  int nbits = modulus.front().BitSize();
+  int length = BITSIZE_WORD(nbits) * 4;
+  nbits = 8 * length;
+
+  unsigned int worksize = base.size();
+  unsigned int nslices = (worksize + batch_size - 1) / batch_size;
+  unsigned int residue = worksize % batch_size;
+
+  batch_size = (nslices > 0) ? batch_size : worksize;
+
+  int thread_id = omp_get_thread_num();
```

Comment:
`int thread_id = omp_get_thread_num();` here is not necessary, `unsigned int buffer_id = 0` is enough, in case the user doesn't use OpenMP for parallelization.